src/olympia/addons/views.py

```diff
@@ -424,20 +424,20 @@ class AddonFeaturedView(AddonSearchView):
     pagination_class = None
 
     def get(self, request, *args, **kwargs):
-        queryset = self.filter_queryset(self.get_queryset())
-        serializer = self.get_serializer(queryset, many=True)
-
         try:
             page_size = int(
                 self.request.GET.get('page_size', api_settings.PAGE_SIZE))
         except ValueError:
             raise exceptions.ParseError('Invalid page_size parameter')
+
         # Simulate pagination-like results, without actual pagination.
-        return Response({'results': serializer.data[:page_size]})
+        queryset = self.filter_queryset(self.get_queryset()[:page_size])
+        serializer = self.get_serializer(queryset, many=True)
+        return Response({'results': serializer.data})
 
     def filter_queryset(self, qs):
         qs = super().filter_queryset(qs)
-        qs = qs.query(query.Bool(must=[Q('term', is_recommended=True)]))
+        qs = qs.query(query.Bool(filter=[Q('term', is_recommended=True)]))
         return qs.query('function_score', functions=[query.SF('random_score')])
```

Comment:
You're applying a random score, but the results aren't ordered by score. Try this:

```diff
diff --git a/src/olympia/addons/views.py b/src/olympia/addons/views.py
index 3b1629fa7d..8381bc59be 100644
--- a/src/olympia/addons/views.py
+++ b/src/olympia/addons/views.py
@@ -423,6 +423,10 @@ class AddonFeaturedView(AddonSearchView):
     # this endpoint since the order is random.
     pagination_class = None
 
+    filter_backends = [
+        ReviewedContentFilter, SearchParameterFilter,
+    ]
+
     def get(self, request, *args, **kwargs):
         queryset = self.filter_queryset(self.get_queryset())
         serializer = self.get_serializer(queryset, many=True)
@@ -438,7 +442,10 @@ def get(self, request, *args, **kwargs):
     def filter_queryset(self, qs):
         qs = super().filter_queryset(qs)
         qs = qs.query(query.Bool(must=[Q('term', is_recommended=True)]))
-        return qs.query('function_score', functions=[query.SF('random_score')])
+        return (
+            qs.query('function_score', functions=[query.SF('random_score')])
+              .sort('_score')
+        )
 
 
 class StaticCategoryView(ListAPIView):
```

Reply:
here you go https://github.com/mozilla/addons-server/pull/13082/commits/0d3366af17375aecdde8b2e79fc2b642bfc57623